src/WWT.Providers/Providers/DSSProvider.cs

```diff
@@ -1,70 +1,58 @@
 using System;
-using System.Configuration;
 using System.IO;
 using WWTWebservices;
 
 namespace WWT.Providers
 {
     public class DSSProvider : RequestProvider
     {
-        public override void Run(WwtContext context)
+        private readonly IPlateTilePyramid _plateTile;
+
+        public DSSProvider(IPlateTilePyramid plateTile)
         {
-            string wwtTilesDir = ConfigurationManager.AppSettings["WWTTilesDir"];
-            string dssTerapixelDir = ConfigurationManager.AppSettings["DssTerapixelDir"];
+            _plateTile = plateTile;
+        }
 
-            string query = context.Request.Params["Q"];
-            string[] values = query.Split(',');
-            int level = Convert.ToInt32(values[0]);
-            int tileX = Convert.ToInt32(values[1]);
-            int tileY = Convert.ToInt32(values[2]);
+        public override void Run(WwtContext context)
+        {
+            var (level, tileX, tileY) = context.GetLevelAndCoordinates();
 
-            int octsetlevel = level;
-            string filename;
+            using (var stream = GetStream(level, tileX, tileY))
+            {
+                if (stream is null)
+                {
+                    context.Response.Write("No image");
+                    context.Response.Close();
+                }
 
+                context.Response.ContentType = "image/png";
+                stream.CopyTo(context.Response.OutputStream);
+                context.Response.Flush();
+                context.Response.End();
+            }
+        }
 
+        private Stream GetStream(int level, int tileX, int tileY)
+        {
             if (level > 12)
             {
-                context.Response.Write("No image");
-                context.Response.Close();
-                return;
+                return null;
             }
-
-            if (level < 8)
+            else if (level < 8)
```

Comment:
Worth keeping in mind here that the DSS dataset is splintered into these sub-plate-files only because of the previous 4 GB file size limit. On the blob storage, the handling here could be simplified to pull data from one large pyramid, rather than splitting between the one file for levels 0-8 (`dssterrapixel.plate`) and the collection of plates for levels 5-12 (`DSSpngL5to12_x{1}_y{2}.plate`).

Relevant here is that I'm pretty sure that our `wwtfiles` blob container *already* contains the loose PNG files as a single 0-12 tile pyramid in the `dsstp` container.

Reply:
Cool. We'll want to ensure the files are in a cannonical format (ie `L{level}X{x-coor}Y{y-coord}.png` so they can be easily found. We can create an issue to track combining that dataset.